### PR TITLE
refactored test/cmd/basicresources to use wrapper functions

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/cc @liggitt @deads2k @smarterclayton 

My search-fu did not help me figure out who originally wrote these lines.

### CODE REVIEW

File: test/cmd/basicresources.sh [hunk 1]
```diff
@@ -6,102 +6,105 @@ set -o pipefail
...
+os::cmd::expect_success 'oc process -f examples/sample-app/application-template-stibuild.json -l name=mytemplate | oc create -f -'
+os::cmd::expect_success 'oc delete all -l name=mytemplate'
+os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world'
+os::cmd::expect_success 'oc get dc/ruby-hello-world'
 
-oc get dc/ruby-hello-world --template='{{ .spec.replicas }}' | grep -q 1
-oc patch dc/ruby-hello-world -p '{"spec": {"replicas": 2}}'
-oc get dc/ruby-hello-world --template='{{ .spec.replicas }}' | grep -q 2
+os::cmd::expect_success_and_text "oc get dc/ruby-hello-world --template='{{ .spec.replicas }}'" '1'
+os::cmd::expect_success "oc patch dc/ruby-hello-world -p '{\"spec\": {\"replicas\": 2}}'"
```
Comment (stevekuznetsov) [inline test/cmd/basicresources.sh]: This isn't the best but if you want both literal single-quotes and double-quotes in a string this is the only way to do it on the same line. We could also declare the patch text as a variable and pass that in.

Comment (liggitt) [inline test/cmd/basicresources.sh]:
```
patch='{"spec":...}'
… "oc patch dc/ruby-hello-world -p '${patch}'"
```
 is probably clearest… does that work?

Comment (stevekuznetsov) [inline test/cmd/basicresources.sh]: Yes, that is exactly what I meant by "declare the patch text". I can do that instead if it's preferable.

File: test/cmd/basicresources.sh [hunk 2]
```diff
@@ -6,102 +6,106 @@ set -o pipefail
...
+os::cmd::expect_success 'oc run --image=openshift/hello-openshift --restart=Never test3'
+os::cmd::expect_success 'oc delete dc/test rc/test2 pod/test3'
+
+os::cmd::expect_success 'oc process -f examples/sample-app/application-template-stibuild.json -l name=mytemplate | oc create -f -'
+os::cmd::expect_success 'oc delete all -l name=mytemplate'
+os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world'
+os::cmd::expect_success 'oc get dc/ruby-hello-world'
+
+os::cmd::expect_success_and_text "oc get dc/ruby-hello-world --template='{{ .spec.replicas }}'" '1'
+patch='{\"spec\": {"replicas": 2}}'
```
Comment (liggitt) [inline test/cmd/basicresources.sh]: Don't need to escape double quotes, right?

**Comment by stevekuznetsov** [comment]
@liggitt Changed to use a envar for the patch text

File: test/cmd/basicresources.sh [hunk 3]
```diff
@@ -6,102 +6,106 @@ set -o pipefail
...
 # This test validates basic resource retrieval and command interaction
 
-[ "$(oc types | grep 'Deployment Configuration')" ]
-[ "$(oc get 2>&1 | grep 'deploymentconfig')" ]
+os::cmd::expect_success_and_text 'oc types' 'Deployment Configuration'
+os::cmd::expect_failure_and_text 'oc get' 'deploymentconfig'
 
 # Test resource builder filtering of files with expected extensions inside directories, and individual files without expected extensions
-[ "$(oc create -f test/fixtures/resource-builder/directory -f test/fixtures/resource-builder/json-no-extension -f test/fixtures/resource-builder/yml-no-extension 2>&1)" ]
+os::cmd::expect_success 'oc create -f test/fixtures/resource-builder/directory'
```
Comment (deads2k) [inline test/cmd/basicresources.sh]: The old command also tested that multiple `-f`s worked together?

File: test/cmd/basicresources.sh [hunk 4]
```diff
@@ -6,102 +6,106 @@ set -o pipefail
...
-oc get secret json-with-extension yml-with-extension
+os::cmd::expect_success 'oc get secret json-with-extension yml-with-extension'
 # Ensure extensionless files inside directories are not processed by resource-builder
-[ "$(oc get secret json-no-extension-in-directory 2>&1 | grep 'not found')" ]
+os::cmd::expect_failure_and_text 'oc get secret json-no-extension-in-directory' 'not found'
 echo "resource-builder: ok"
 
-oc get pods --match-server-version
+os::cmd::expect_success 'oc get pods --match-server-version'
 [ "$(oc create -f examples/hello-openshift/hello-pod.json 2>&1 | grep 'pod "hello-openshift" created')" ]
```
Comment (deads2k) [inline test/cmd/basicresources.sh]: Why was this not updated?

File: test/cmd/basicresources.sh [hunk 5]
```diff
@@ -6,102 +6,106 @@ set -o pipefail
...
+os::cmd::expect_success 'oc get services'
+os::cmd::expect_success 'oc create -f test/integration/fixtures/test-service.json'
+os::cmd::expect_success 'oc delete services frontend'
 echo "services: ok"
 
-oc create -f test/fixtures/mixed-api-versions.yaml
-oc get    -f test/fixtures/mixed-api-versions.yaml -o yaml > /dev/null
-oc delete -f test/fixtures/mixed-api-versions.yaml
+os::cmd::expect_success 'oc create -f test/fixtures/mixed-api-versions.yaml'
+os::cmd::expect_success 'oc get    -f test/fixtures/mixed-api-versions.yaml -o yaml > /dev/null'
```
Comment (deads2k) [inline test/cmd/basicresources.sh]: I shouldn't care about redirecting anymore right?

File: test/cmd/basicresources.sh (hunk 3 above)
Comment (stevekuznetsov) [inline test/cmd/basicresources.sh]: This is true.

**Comment by stevekuznetsov** [comment]
@deads2k addressed comments

**Comment by deads2k** [comment]
lgtm [merge]

**Pull request merged**